ext/intl/formatter/formatter_format.c

```diff
@@ -103,9 +103,15 @@ PHP_FUNCTION( numfmt_format )
 			}
 			INTL_METHOD_CHECK_STATUS( nfo, "Number formatting failed" );
 			break;
-
+		case FORMAT_TYPE_CURRENCY:
+			if (getThis()) {
+				zend_argument_value_error(2, "cannot use NumberFormatter::TYPE_CURRENCY constant, use formatCurrency() method instead");
```

Comment:
Not sure about the `formatCurrency()`; shouldn't that better be `::formatCurrency()`? cc @kocsismate

Reply:
My preference would be to prepend the class name before the method name, just like in case of the constant name.

Reply:
Otherwise `Argument #2 ($type) cannot use NumberFormatter::TYPE_CURRENCY...` sounds a bit unnatural to me. What about `Argument #2 ($type) cannot be NumberFormatter::TYPE_CURRENCY...`?

Reply:
> What about `Argument #2 ($type) cannot be NumberFormatter::TYPE_CURRENCY...`?

Yeah, that's better, in my opinion.

And maybe we should print the actual class name, since `NumberFormatter` is not final, nor are its class constants.

Reply:
ACK